Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,14 +48,58 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            pypircPath = tl.getVariable("PYPIRC_PATH");
+            const pypirc = fs.readFileSync(pypircPath, 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let reposList: string[] = [];
+
+            for (let entry of newEndpointsToAdd) {
+
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
+
+                if (entry.packageSource.feedName in fileContent){
+                    tl.warning(tl.loc("DuplicateRegistry", entry.packageSource.feedName));
+                    if (internalFeed.includes(entry)) {
+                        internalFeed.pop();
+                        continue;
+                    }
+                    externalEndpoints.pop();
+                    continue;
+                }
+
+                let repo = new Repository(
+                    entry.packageSource.feedName,
+                    entry.packageSource.feedUri,
+                    entry.username,
+                    entry.password
+                );
+
+                reposList.push(repo.toString() + `${os.EOL}`);
+
+                fileContent["distutils"]["index-servers"] += " " + entry.packageSource.feedName;
+            }
+
+            let encodedStr = ini.encode(fileContent);
+            fs.writeFileSync(pypircPath, encodedStr);
+
+            fs.appendFileSync(pypircPath, `${os.EOL}`, 'utf8');
+
+            for (let repo of reposList) {
+                fs.appendFileSync(pypircPath, repo, 'utf8');
+            }
+        }
 
         // Configuring the pypirc file
         internalFeedSuccessCount = internalFeed.length;
         externalFeedSuccessCount = externalEndpoints.length;
-        console.log(tl.loc("Info_SuccessAddingAuth", internalFeedSuccessCount, externalFeedSuccessCount));
+        tl.loc("Info_SuccessAddingAuth", internalFeedSuccessCount, externalFeedSuccessCount);
```

Comment:
I think you still need the console.log to log the tl.loc string to the output. Alternatively, you can use tl.Info I think

Reply:
Added this back!